### PR TITLE
DomEvent.DoubleTap ignores clicks on <label>s

### LINKED ISSUE #8225: Firefox 100: Clicking control fires ghost dblclick
**Steps to reproduce**
- Click the control in the top right corner with Leaflet 1.7.1 activated
- Click the control in the top right corner with Leaflet 1.8.0 activated

**Expected behavior**
The map has a button in the top right corner and a text area in the bottom left. Clicking the button should toggle the area's visibility.

**Current behavior**
Clicking the button not only toggles the area's visibility, but also zooms in on the map. This only happens in Firefox, not in Chrome, since upgrading to leaflet 1.8.0.

**Environment**
- Leaflet version: 1.8.0
- Browser (with version): Firefox 100.0
- OS/Platform (with version): Windows 10

**Minimal example reproducing the issue**
~~See this JSFiddle: https://jsfiddle.net/petergrondelaers/pvk4hxmw/~~

https://jsfiddle.net/51bLtj9x/ or https://run.plnkr.co/preview/cl30e37zy0007206kqu0it63k/

### PR DESCRIPTION
**Pull request by IvanSanchez**
Fixes #8225.

When clicking on a `<label>` linked to an `<input>` (or vice versa), browsers trigger an extra click on the related element, as per https://html.spec.whatwg.org/multipage/forms.html#the-label-element . This causes the `DomEvent.DoubleTap.js` logic to activate, and fire a ghost `dblclick` event.

So after sleeping on the problem, I found what should be a reliable approach: if the problem is the extra clicks that browsers introduce on `<label>`s... ignore those! Get the propagation path of the event, and check if:
- The event happened on a `<label>`, or on a child of a `<label>`
- The `<label>` has a `for` attribute (thus ignoring standalone labels and nested labels)
- The event is not happening on an `<input>` or child of an `<input>` (which means a nested label)

I think this approach is sound, but I haven't tested it extensively (or, to be honest, at all). I'm assuming that browsers do *not* trigger an extra click when the `<input>` is nested within a `<label>`, but I might be wrong about that.

Do note that `composedPath` is not implemented by all browsers (hence the `if (e.composedPath)` branch), so... (*checks notes*)... IE, Edge<74 and Safari<=9 would still display the buggy behaviour of #8225.

cc @pgrondelaers

### CODE REVIEW

File: src/dom/DomEvent.DoubleTap.js [hunk 1]
```diff
@@ -44,6 +44,27 @@ export function addDoubleTapListener(obj, handler) {
...
 
+		// When clicking on an <input>, the browser generates a click on its
+		// <label> (and vice versa) triggering two clicks in quick succession.
+		// This ignores clicks on elements which are a label with a 'ref'
+		// attribute (or children of such a label), but not children of
+		// a <input>.
+		if (e.composedPath) {
+			var path = e.composedPath();
+			if (path.some(function (el) {
+				return el instanceof HTMLLabelElement && el.attributes.ref;
```
Comment (Malvoz) [inline src/dom/DomEvent.DoubleTap.js:55]: Is `ref` supposed to be [`for`](https://html.spec.whatwg.org/multipage/forms.html#attr-label-for)?

Comment (IvanSanchez) [reply src/dom/DomEvent.DoubleTap.js]: ...and that's why I should be fully awake before sending PRs.

Comment (Falke-Design) [reply src/dom/DomEvent.DoubleTap.js]: Then the following is wrong too?
> The <label> has a `ref` attribute (thus ignoring standalone labels and nested labels)

**Comment by Falke-Design** [comment]
Can we use `Event.path` instead of `composedPath`? Also see: #8226

**Comment by Malvoz** [comment]
> Can we use `Event.path` instead of `composedPath`? Also see: #8226

Would that result in a compatibility issue?

from https://chromestatus.com/feature/5726124632965120#comments:

> Existing pages may use the following polyfill to keep functions:
> 
> ```js
> if (!Event.prototype.path) {
>   Object.defineProperty(Event.prototype, 'path', {
>     get() { return this.composedPath(); }
>   });
> }
> ```

**Comment by IvanSanchez** [comment]
> Can we use `Event.path` instead of `composedPath`? Also see: #8226

And do something like `var path = (e.composedPath && e.composedPath()) || e.path || []`? 

I think the answer is "not really". According to a comment on https://github.com/tinymce/tinymce/issues/4490 , it's chrome-specific, not IE-specific.

Might as well polyfill as

```js
L.DomEvent.getEventPath = function (ev) {
  if (ev.composedPath) {
    return ev.composedPath();
  }

  var path = [];
  var el = ev.target;

  while (el) {
    path.push(el);
    el = el.parentNode;
  }
  return path;
};
```

**Review by Falke-Design — APPROVED** [review]
@IvanSanchez I added also an test, please take a look on it before merging

**Pull request merged**